content/blog/2023/09/03/2023-09-03-jsr305-probe-blog.adoc

```diff
@@ -0,0 +1,52 @@
+---
+layout: post
+title: JSR-305 Probe Blog
+tags:
+- gsoc
+- gsoc2023
+- healthscore
+- probes
+- plugin
+authors:
+- jagruti
+opengraph:
+  image: /images/gsoc/2023/jagruti/jsr-305-probe-image.png
+sig:
+- gsoc
+---
+
+== Introduction
+
+The `JSR-305` framework provides annotations to detect software defects. Its executive committee voted it as dormant since May 2012 (see: link:https://jcp.org/en/jsr/detail?id=305[JSR 305: Annotations for Software Defect Detection
+])
+
+It is important to stop using the framework for the following reasons:
+
+- It is not maintained anymore.
+- The annotations have questionable license.
+- The annotations are in the reserved javax namespace and there is no public release of the spec (Nor is there ever likely to be. See point 1).
+(Source: link:https://groups.google.com/g/jenkinsci-dev/c/uE1wwtVi1W0/m/gLxdEJmlBQAJ[Jenkins Developers Discussion])
+
+
+`@Nonnull` and `@CheckForNull` annotations (part of JSR-305) have been link:https://www.jenkins.io/doc/developer/tutorial-improve/replace-jsr-305-annotations/[deprecated since 2016].
+
+The `javax.annotation.Nonnull` import should be replaced with `edu.umd.cs.findbugs.annotations.NonNull`.
```

Comment:
Why giving this example and not the replacement for `CheckForNull`? Not sure it is very important to have here?